### PR TITLE
[MODULAR] Cleans up dorm vendor code

### PR DESCRIPTION
**Pull request by softcerv**
## About The Pull Request
Cleans up the code for the dorm vendor. Adds in autodoc comments and implements early returns.

## How This Contributes To The Skyrat Roleplay Experience
good code is HRP

## Changelog

:cl:
code: cleans up dorm vendor code
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_machinery/lustwish.dm [hunk 1]
```diff
@@ -129,64 +132,71 @@
...
+	user.visible_message(span_boldnotice("Something changes in [src] with a loud clunk."))
+	card_used = !card_used
+
+	if(card_used)
+		default_price = 0
+		extra_price = 0
+
+		return
+
+	default_price = 30
```
Comment (larentoun) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_machinery/lustwish.dm]: Why not use `initial()`?

**Review by Zonespace27 — CHANGES_REQUESTED** [review]
Just do the initial() thing and lgtm

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_machinery/lustwish.dm [hunk 2]
```diff
@@ -129,64 +132,71 @@
...
+	user.visible_message(span_boldnotice("Something changes in [src] with a loud clunk."))
+	card_used = !card_used
+
+	if(card_used)
+		default_price = 0
+		extra_price = 0
+
+		return
+
+	default_price = initial(src.default_price)
```
Comment (Zonespace27) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_machinery/lustwish.dm]:
```suggestion
	default_price = initial(default_price)
```

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_machinery/lustwish.dm [hunk 3]
```diff
@@ -129,64 +132,71 @@
...
+	extra_price = initial(src.extra_price)
```
Comment (Zonespace27) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_machinery/lustwish.dm]:
```suggestion
	extra_price = initial(extra_price)
```

**Review by Zonespace27 — APPROVED** [review]

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_machinery/lustwish.dm [hunk 4]
```diff
@@ -129,64 +132,71 @@
...
+
+	if(!card_used)
 		return
 
+	var/choice = show_radial_menu(user, src, vend_designs, custom_check = CALLBACK(src, .proc/check_menu, user, used_item), radius = 50, require_near = TRUE)
+	if(!choice)
+		return FALSE
+
+	current_color = choice
+	update_icon()
```
Comment (tf-4) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_machinery/lustwish.dm]:
```suggestion
	if(choice)
		current_color = choice
		update_icon()
```
No need to early return here and I feel it's more readable this way

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_machinery/lustwish.dm [hunk 5]
```diff
@@ -129,64 +132,71 @@
...
+
+///Preforms checks to see if the user can change the color on the vending machine.
```
Comment (tf-4) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_machinery/lustwish.dm]:
```suggestion
///Performs checks to see if the user can change the color on the vending machine.
```

**Review by tf-4 — CHANGES_REQUESTED** [review]

**Review by tf-4 — APPROVED** [review]

**Pull request merged**